neural_compressor/adaptor/onnxrt.py

```diff
@@ -186,29 +186,38 @@ def smooth_quant(self, model, dataloader, iterations, alpha=0.5, folding=True,
             return self.smooth_quant_model
 
         from .ox_utils.smooth_quant import ORTSmoothQuant
-        # TODO remove quantize_config as it no consumer
-        quantize_config = None
-        # pre-optimization -> sq
+
+        # set params to cur_sq_args
+        self.cur_sq_args['alpha'] = alpha
+        self.cur_sq_args['folding'] = folding
+        self.cur_sq_args['percentile'] = percentile
+        self.cur_sq_args['op_types'] = op_types
+        self.cur_sq_args['scales_per_op'] = scales_per_op
+        self.cur_sq_args['calib_iter'] = iterations
+        
+        # pre-optimization
         self._pre_optimize(model)
+
         # assign the algo to the adaptor, so adaptor can call it later when needed
         self.sq = ORTSmoothQuant(self.pre_optimized_model, dataloader, self.reduce_range, self.backend)
-        self.smooth_quant_model = self.sq.transform(
-            alpha, folding, percentile, op_types, scales_per_op, iterations, quantize_config)
+        self.sq.record_max_info = record_max_info
+        self.smooth_quant_model = self.sq.transform(**self.cur_sq_args)
         logger.info("Updated the pre-optimized model with smooth quant model.")
         # TODO double-check the smooth_quant_model and pre_optimized_model to make sure there no two fp32 model replicas
         self.pre_optimized_model = self.smooth_quant_model
         return self.smooth_quant_model
     
     def _need_smooth_quant(self, tune_cfg) -> bool:
-        # compare the alpha from tune_cfg and current alpha to decide whether re-smooth model or not
-        # TODO
-        return False
-    
-    def _parse_sq_args(self, tune_cfg, cur_sq_args) -> Dict:
-        # parse the sq args according to the tune cfg and current sq args
-        # TODO
-        return {}
-        
+        """Check the model needs smooth quant or not."""
+        recipe_cfgs = tune_cfg.get('recipe_cfgs', None)
+        if recipe_cfgs and recipe_cfgs.get('smooth_quant', False) \
+            and recipe_cfgs['smooth_quant_args'].get('alpha', None):
+            # update alpha according to tune_cfg
+            self.cur_sq_args['alpha'] = \
+                tune_cfg['recipe_cfgs']['smooth_quant_args']['alpha']
+            return True
```

Comment:
The `need_smooth_quant` condition should be: the `alpha` from `tune_cfg` is not `None` and not equal to the `alpha` in `cur_sq_args`.

Reply:
After alpha tuning, the best alpha might equal to the last tuned alpha.

For example: baseline acc = 1.0, alpha = [0.1, 0.2]
- alpha = 0.1, acc = 0.8 --> alpha = 0.2, acc = 0.9 -(end alpha tune)-> other tuning with best alpha = 0.2 which also needs excute smooth quant

Reply:
I see, thx.